e2e/config/azure.go

```diff
@@ -324,10 +359,105 @@ func (a *AzureClient) assignReaderRoleToBlobStorage(ctx context.Context, princip
 		if errors.As(err, &respError) && respError.StatusCode == http.StatusConflict {
 			return nil
 		}
-		return fmt.Errorf("assign reader role: %w", err)
+		return fmt.Errorf("assign Storage Blob Data Contributor role: %w", err)
 	}
 	return nil
+}
+
+func (a *AzureClient) LatestSIGImageVersionByTag(ctx context.Context, image *Image, tagName, tagValue string) (VHDResourceID, error) {
```

Comment:
nit: `GetLatestSIGImageVersionByTag`

Reply:
I don't like `Get/Set`. Feels redundant to me. Technically half of the methods can have this prefix.